code/datums/gamemode/objectives/target/assassinate.dm

```diff
@@ -99,25 +99,26 @@ var/list/assassination_objectives = list()
 		var/datum/component/uplink/enemy_uplink = target.find_syndicate_uplink(enemy.uplink)
 		//chances are the target's uplink is no longer on their mind.current especially if they got decapitated or such.
 		//by associating the uplink with the role we can at least try and get the TCs out of it.
+		//Increases the victor's uplink (if it finds it) by 8tcs, or the number in the victim's uplink, whichever was greater, and sets the victim's to 0 (if it finds it).
 
 		if (A.target == owner)
-			to_chat(owner.current, "<span class='notice'>The Syndicate congratulates you on your Victory. Look forward to be assigned on higher risk operations another day.</span>")
+			to_chat(owner.current, "<span class='notice'>The Syndicate congratulates you on your victory. Look forward to be assigned on higher risk operations another day.</span>")
 		else
 			if (owner_uplink)
-				owner_uplink.telecrystals += DOUBLE_AGENT_TC_REWARD
-				if (enemy_uplink)
-					owner_uplink.telecrystals += enemy_uplink.telecrystals
-					enemy_uplink.telecrystals = 0
-				to_chat(owner.current, "<span class='notice'>Good work agent. [DOUBLE_AGENT_TC_REWARD] additional tele-crystals have been sent to your uplink.</span>")
+				owner_uplink.telecrystals += max(reward, enemy_uplink.telecrystals)
```

Comment:
`owner_uplink.telecrystals += max(reward, enemy_uplink?.telecrystals)`
the `?` is important

Reply:
would you mind helping me understand why? i tested the change and it works without it

Reply:
the `?` means it'll check if that thing actually exists - if you arrest the use of a var when that var doesn't exist, you'll get runtimes

Reply:
gotcha, thanks